plugin/cassandra/plugin.go

```diff
@@ -0,0 +1,666 @@
+
+type CassandraInfo struct {
+	Host     string
+	Port     string
+	User     string
+	Password string
+	Keyspace string
+	BinDir   string
+	DataDir  string
+	Tar      string
+}
+
+// This function should be used to return the plugin's PluginInfo, however you decide to implement it
+func (p CassandraPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+// Called to validate endpoints from the command line
+func (p CassandraPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValueDefault("cassandra_host", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_host          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_host}          using default node @C{%s}\n", DefaultHost)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_host}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_port", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_port          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_port}          using default port @C{%s}\n", DefaultPort)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_port}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_user", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_user          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_user}          using default user @C{%s}\n", DefaultUser)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_user}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_password", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_password      %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_password}      using default password @C{%s}\n", DefaultPassword)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_password}      @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("cassandra_keyspace")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_keyspace      %s}\n", err)
+		fail = true
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_keyspace}      @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_bindir", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_bindir        %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_bindir}        using default @C{%s}\n", DefaultBinDir)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_bindir}        @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_datadir", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_datadir       %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_datadir}       using default @C{%s}\n", DefaultDataDir)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_datadir}       @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_tar", "")
```

Comment:
Similar to the comments on `cassandra_bindir`, we can just remove this outright and just exec `tar`.  Given that this plugin doesn't need anything special à la bsdtar, standard GNU tar in `/bin` should suffice.

Reply:
Yep, no problem.